src/Microsoft.AspNet.OData.Shared/Common/ODataPathHelper.cs

```diff
@@ -0,0 +1,71 @@
+//-----------------------------------------------------------------------------
+// <copyright file="ODataPathHelper.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved. 
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using Microsoft.OData.UriParser;
+
+namespace Microsoft.AspNet.OData.Common
+{
+    /// <summary>
+    /// Helper methods for <see cref="ODataPath"/>.
+    /// </summary>
+    public static class ODataPathHelper
+    {
+        /// <summary>
+        /// Get the keys from a <see cref="KeySegment"/>.
+        /// </summary>
+        /// <param name="keySegment">The <see cref="KeySegment"/> to extract the keys.</param>
+        /// <returns>Dictionary of keys.</returns>
+        public static Dictionary<string, object> GetKeysFromKeySegment(KeySegment keySegment)
```

Comment:
Could this have been an extension method in `ODataPathExtensions`?
```csharp
public static IDictionary<string, object> GetKeys(this KeySegment keySegment)
```

Reply:
This is more of `KeySegmentExtensions`. That's why I didn't add it in the `ODataPathExtensions`